cmake/AwsSharedLibSetup.cmake

```diff
@@ -19,6 +21,12 @@ if (UNIX AND NOT APPLE)
 endif()
 
 function(aws_prepare_shared_lib_exports target)
+    if (VERSION_LIBS)
+        # Our ABI is not yet stable
+        set_target_properties(${target} PROPERTIES VERSION 1.0.0)
+        set_target_properties(${target} PROPERTIES SOVERSION 0unstable)
```

Comment:
`option(VERSION_LIBS ...)` belongs in the `AwsSharedLibSetup.cmake`, but these lines should probably still be in aws-c-common's `CMakeLists.txt`. That way, each library can set its own version# (when VERSION_LIBS is ON)

Reply:
Having to remember to put it in each lib defeats the point

Reply:
Yeah but do we really want to force all of our libs to always use the same version number? I'm ok with putting SOVERSION here, but I do think the VERSION should be lib specific.